posts/2022-08-04-jakarta-core-profile-22009-beta.adoc

```diff
@@ -0,0 +1,270 @@
+
+* The Liberty configuration element has been superseded by a new element that applies to all versions of CDI (1.2 and newer).
+
+[source, xml]
+----
+<cdi enableImplicitBeanArchives="false" emptyBeansXmlCDI3Compatibility="true"/>
+----
+
+* The attribute `enableImplicitBeanArchives` is the same as it was in previous versions.
+If set to `true` (default) then archives which do not contain a `beans.xml` file will be treated as Implicit Bean Archives and will be scanned for classes that have Bean Defining Annotations.
+If set to `false` then archives which do not contain a `beans.xml` file will be not be scanned for annotated classes.
+
+* The attribute `emptyBeansXmlCDI3Compatibility` only applies to CDI 4.0.
+If set to `true` then an archive which contains an empty `beans.xml` file will be treated as an Explicit Bean Archive (as it was in CDI 3.0 and earlier)
+
+* If set to `false` (default) then an archive which contains an empty `beans.xml` file will be treated as an Implicit Bean Archive. This is the equivalent to setting `bean-discovery-mode="annotated"`.
+
+[source, xml]
+----
+<featureManager>
+  <feature>cdi-4.0</feature>
+</featureManager>
+
+<cdi enableImplicitBeanArchives="false" emptyBeansXmlCDI3Compatibility="true"/>
+----
+
+[#expression]
+== Jakarta EE 10 Expression Language 5.0
+
+The `expressionLanguage-5.0` feature is an implementation of the Expression Language 5.0 specification for Jakarta EE 10. The Expression Language 5.0 includes a number of new features and specification clarifications.
+
+To learn about the changes between the Expression Language 4.0 and Expression Language 5.0, reference the  link:https://jakarta.ee/specifications/expression-language/5.0/jakarta-expression-language-spec-5.0.html#changes-between-5-0-and-4-0[Expression Language 5.0 spec documentation].
+
+One of the new features is the addition of the link:https://jakarta.ee/specifications/expression-language/5.0/apidocs/jakarta.el/jakarta/el/methodreference[MethodReference] which "Provides information about the method to which a method expression resolves." This new class allows developers to access any annotations present on a particular method using the `getAnnotations()` method. There are also a few deprecations and clarifications within the Expression Language 5.0 Specification.
+
+[source,xml]
+----
+   <featureManager>
+        <feature>expressionLanguage-5.0</feature>
+    </featureManager>
+----
+
+More information regarding Jakarta Expression Language 5.0 can be  found on the 
+link:https://jakarta.ee/specifications/expression-language/5.0/[jakarta.ee site].
+
+[#microprofile]
+== MicroProfile OpenAPI 3.1 
+
+OpenAPI is a standardised way of documenting REST APIs in a JSON or YAML format. MicroProfile OpenAPI helps you generate and serve OpenAPI documentation for your REST applications built using JAX-RS or Jakarta restfulWS. This can be useful for developers to test out the API during development, or for people using the API in production.
```

Comment:
"Jakarta restfulWS"?

Reply:
Sorry I've used the feature short name by mistake, it should be the full name for the spec: "Jakarta RESTful Web Services"

```suggestion
OpenAPI is a standardised way of documenting REST APIs in a JSON or YAML format. MicroProfile OpenAPI helps you generate and serve OpenAPI documentation for your REST applications built using JAX-RS or Jakarta RESTful Web Services. This can be useful for developers to test out the API during development, or for people using the API in production.
```